### PR TITLE
Drop dummy copy constructors in CPrivateSend*Session

### PR DESCRIPTION
**Pull request by UdjinM6**
See individual commits. Replaces #2304

### CODE REVIEW

File: src/privatesend-client.cpp [hunk 1]
```diff
@@ -82,7 +82,7 @@ void CPrivateSendClientManager::ProcessMessage(CNode* pfrom, const std::string&
                 }
             }
         } else {
-            LOCK(cs_vecsessions); // have to lock this first to avoid deadlocks with cs_vecqueue
+            LOCK(cs_deqsessions); // have to lock this first to avoid deadlocks with cs_vecqueue
```
Comment (nmarley) [inline src/privatesend-client.cpp]: nit: `cs_vecqueue` still in comment

Comment (UdjinM6) [reply src/privatesend-client.cpp]: Hmmm... and...? vecqueue != vecsession :)

Comment (nmarley) [reply src/privatesend-client.cpp]: You got me there. 😳

File: src/privatesend-client.h [hunk 2]
```diff
@@ -204,7 +203,7 @@ class CPrivateSendClientManager : public CPrivateSendBaseManager
     CPrivateSendClientManager() :
         vecMasternodesUsed(),
         vecDenominationsSkipped(),
-        vecSessions(),
+        deqSessions(),
```
Comment (codablock) [inline src/privatesend-client.h]: I'm wondering why all these containers are explicitly default initialized here?

File: src/privatesend.h [hunk 3]
```diff
@@ -339,7 +339,6 @@ class CPrivateSendBaseSession
         finalMutableTransaction(),
         nSessionDenom(0)
         {}
-    CPrivateSendBaseSession(const CPrivateSendBaseSession& other) { /* dummy copy constructor*/ SetNull(); }
```
Comment (codablock) [inline src/privatesend.h]: Oh my :see_no_evil: 
Didn't even notice this one, good it's going away now :)

Comment (codablock) [reply src/privatesend.h]: I'm wondering this didn't cause problems due to the use of std::vector before. Shouldn't this have resetted all sessions from time to time?

Comment (UdjinM6) [reply src/privatesend.h]: Yep, good point, but I think it was only the case on mixing start when we were pushing new sessions into the vector until vector had the desired size.

**Pull request merged**